tests/test_action_checks.py

```diff
@@ -0,0 +1,98 @@
+            ("Test6_OtterBye", "Critical hotfix otter-bye emergency deploy", False),
+            ("Test7_MultipleFlags", "Infrastructure update no-review,skip-otter,otter-restricted", False),
+            ("Test8_StandardFeature", "Add user management features and improve UI", True),
+            ("Test9_ConventionalCommit", "feat: implement new logging system", True),
+            ("Test10_BugFix", "fix: resolve authentication bug", True),
+            ("Test11_Documentation", "docs: update API documentation", True),
+            ("Test12_Maintenance", "chore: upgrade dependencies", True),
+            ("Test13_Testing", "test: add integration tests", True),
+            ("Test14_CodeRefactor", "refactor: optimize database queries", True),
+            ("Test15_Formatting", "style: format code according to standards", True),
+            ("Test16_Performance", "perf: improve loading times", True),
+            ("Test17_NoSkipFlags", "This PR skips nothing", True),
+        ]
+
+        print("==============================")
+        print("Running PR Title Check Tests")
+        print("==============================")
+
+        for description, title, expected in test_cases:
+            with self.subTest(description=description):
+                print(f"Running {description} with title: '{title}'")
+                result = self.check_review_requested(title)
+                print(f"Expected result: {expected}")
+                
+                try:
+                    self.assertEqual(result, expected, 
+                        f"{description} failed: expected {expected}, got {result}")
+                    print("✅ Passed")
+                except AssertionError as e:
+                    print("❌ Failed")
+                    print(str(e))
+                    sys.exit(1)
+                print("---------------------------")
+
+    def test_pr_states(self):
+        test_cases = [
+            ("TestA_merged", "merged", False),
+            ("TestB_open", "open", True),
+            ("TestC_closed", "closed", False),
+        ]
+
+        print("==============================")
+        print("Running PR State Check Tests")
+        print("==============================")
+
+        for description, state, expected in test_cases:
+            with self.subTest(description=description):
+                print(f"Running {description} with state: '{state}'")
+                result = self.check_pr_merged(state)
+                print(f"Expected result: {expected}")
+                
+                try:
+                    self.assertEqual(result, expected,
+                        f"{description} failed: expected {expected}, got {result}")
+                    print("✅ Passed")
+                except AssertionError as e:
+                    print("❌ Failed") 
+                    print(str(e))
+                    sys.exit(1)
+                print("---------------------------")
+
+if __name__ == '__main__':
```

Comment:
✅ Good practice to include this block to run tests when the script is executed directly. 👍